File: solutions/Java/[0977_Easy] Squares of a Sorted Array/sortedSquares.java
class Solution {
  public int[] sortedSquares(int[] nums) {
    int i = 0;
    int j = nums.length - 1;
    int[] result = new int[nums.length];

    for (int k = nums.length - 1; k >= 0; k--) {
      if (Math.abs(nums[i]) >= Math.abs(nums[j])) {
        result[k] = (int) Math.pow(nums[i], 2);
        i++;
      } else {
        result[k] = (int) Math.pow(nums[j], 2);
        j--;
      }
    }

    return result;
  }
}

File: solutions/TypeScript/[0977_Easy] Squares of a Sorted Array/sortedSquares.ts
function sortedSquares(nums: number[]): number[] {
  let i = 0;
  let j = nums.length - 1;
  const result: number[] = [];

  for (let k = nums.length - 1; k >= 0; k--) {
    if (Math.abs(nums[i]) >= Math.abs(nums[j])) {
      result[k] = Math.pow(nums[i], 2);
      i++;
    } else {
      result[k] = Math.pow(nums[j], 2);
      j--;
    }
  }

  return result;
}
